functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,691 @@
+package gatekeeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testClient "github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	gkv1alpha "github.com/open-policy-agent/gatekeeper/apis/mutations/v1alpha1"
+)
+
+const (
+	// TestingNamespace is the namespace for resources in this test
+	TestingNamespace = "gatekeeper-testing"
+)
+
+var _ = Describe("gatekeeper", func() {
+	client := testClient.Client
+
+	AfterEach(func() {
+		err := deletePods(TestingNamespace, client)
+		Expect(err).NotTo(HaveOccurred())
+
+		namespacesUsed := []string{"mutation-included", "mutation-excluded", "gk-test-object"}
+
+		for _, namespace := range namespacesUsed {
+			if namespaces.Exists(namespace, client) {
+				err := client.Namespaces().Delete(context.Background(), namespace, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}
+		}
+
+		for _, namespace := range namespacesUsed {
+			if namespaces.Exists(namespace, client) {
```

Comment:
We can skip `if namespaces.Exists(namespace, client)`, this is done in WaitForDeletion